packages/prop-controllers/src/gap-x/gap-x.ts

```diff
@@ -1,102 +1,16 @@
-import { match } from 'ts-pattern'
-import { ControlDataTypeKey, Options, Types } from '../prop-controllers'
-import { z } from 'zod'
-import { GapData, ResponsiveGapData, responsiveGapDataSchema } from '../data'
+import { Types } from '../prop-controllers'
+import { GapData, responsiveGapDataSchema } from '../data'
+import { versionedPropDef } from '../versioned'
 
-const gapXPropControllerDataV0Schema = responsiveGapDataSchema
-
-export type GapXPropControllerDataV0 = z.infer<
-  typeof gapXPropControllerDataV0Schema
->
-
-export const GapXPropControllerDataV1Type = 'prop-controllers::gap-x::v1'
-
-const gapXPropControllerDataV1Schema = z.object({
-  [ControlDataTypeKey]: z.literal(GapXPropControllerDataV1Type),
-  value: responsiveGapDataSchema,
-})
-
-export type GapXPropControllerDataV1 = z.infer<
-  typeof gapXPropControllerDataV1Schema
->
-
-export const gapXPropControllerDataSchema = z.union([
-  gapXPropControllerDataV0Schema,
-  gapXPropControllerDataV1Schema,
-])
-
-export type GapXPropControllerData = z.infer<
-  typeof gapXPropControllerDataSchema
->
-
-export type GapXOptions = Options<{
-  preset?: ResponsiveGapData
+export const GapX = versionedPropDef<{
+  preset?: unknown
```

Comment:
If present in "raw" options, the `preset` prop will recieve the correct type as a part of the property definition "construction" here.